app/src/main/java/com/example/concordia_campus_guide/Activities/PathsActivity.java

```diff
@@ -130,19 +130,18 @@ public void checkFromToType(Place from, Place to) {
 
     public void setIndoorPaths() {
         if(!fromIsIndoor && !toIsIndoor) return;
-        String floorCode;
-        Place tempFrom = from;
-        Place tempTo = to;
 
-        if(fromIsIndoor && !toIsIndoor){
-            floorCode = ((RoomModel) from).getFloorCode();
-            tempTo = new RoomModel(from.getCenterCoordinates(), floorCode.substring(0, floorCode.indexOf('-')), floorCode);
+        //Outdoor -> Indoor
+        if (!fromIsIndoor) {
+            locationFragment.setIndoorPaths(mViewModel.getEntrance(to), to);
         }
-        if(toIsIndoor && !fromIsIndoor){
-            floorCode = ((RoomModel) to).getFloorCode();
-            tempFrom = new RoomModel(to.getCenterCoordinates(), floorCode.substring(0, floorCode.indexOf('-')), floorCode);
+        else if(mViewModel.arePlacesSeparatedByATunnel(from, to)){
+            locationFragment.setIndoorPaths(from, to);
+        }
+        //[from -> from_entrance ] + outdoor directions + [to_entrance -> to]
+        else if (toIsIndoor && fromIsIndoor) {
```

Comment:
I think adding a check here for when the to-from are from the same building would fix the H-8 weird line problem.